app/controllers/taxonomy_signups_controller.rb

```diff
@@ -0,0 +1,43 @@
+class TaxonomySignupsController < ApplicationController
+  def new
+    redirect_to '/' and return unless params[:paths].present?
```

Comment:
That would break in the absence of a referer - redirecting to root is the simplest thing to do in this case, especially as you'd never have a link on the site without the paths specified.